backends/dpdk/dpdkHelpers.cpp

```diff
@@ -786,24 +786,49 @@ bool ConvertStatementToDpdk::preorder(const IR::MethodCallStatement *s) {
     return false;
 }
 
-// This function assumes EliminateSwitch midend pass is applied and only handles
-// action_run in switch expression.
+// If we get a switch statement with constant switch expression, only emit the body of
+// matching case label
+bool ConvertStatementToDpdk::handleConstSwitch(const IR::SwitchStatement *s) {
+    auto constSwitchExpr = s->expression->to<IR::Constant>();
+    bool constSwitchFlag = false;
```

Comment:
Just for safety I would add here a `CHECK_NULL(constSwitchExpr)`.
In general, you should check invariants that must be ensured by callers, because code may be refactored. Alternatively, you could have here the check for Constant and return right away if it is not constant. I think that's a better design.